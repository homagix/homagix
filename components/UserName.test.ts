import { describe, it, expect } from "vitest"
import { setup } from "@nuxt/test-utils"
import { mountSuspended } from "@nuxt/test-utils/runtime"
import UserName from "~/components/UserName.client.vue"
import { mockNuxtImport } from "@nuxt/test-utils/runtime"
import { flushPromises } from "@vue/test-utils"

mockNuxtImport("useCurrentUser", () => {
  return () => ({ currentUser: { firstName: "John" } })
})

describe("UserName", async () => {
  setup({ server: false })
  it("should render as expected", async () => {
    const wrapper = await mountSuspended(UserName)
    await flushPromises()
    expect(wrapper.html()).toMatchSnapshot()
  })
})
