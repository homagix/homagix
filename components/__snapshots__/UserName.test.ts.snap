// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserName > should render as expected 1`] = `"<span class="username">John</span>"`;
